processing/src/main/java/io/druid/segment/DimensionHandlerUtil.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.base.Function;
+import com.metamx.common.IAE;
+import io.druid.segment.column.ColumnCapabilities;
+import io.druid.segment.column.ValueType;
+
+public final class DimensionHandlerUtil
+{
+  private DimensionHandlerUtil() {}
+
+  public static DimensionHandler getHandlerFromCapabilities(String dimensionName, ColumnCapabilities capabilities)
```

Comment:
This feels like a major pinch point for extensibility. Meaning there is no discovery capabilities here for non-default dimension handling.

would it be possible to either use a guice multi-map binding, or a Jackson factory method with something like 

`objectMapper.convertValue(ImmutableMap.of("type", capabilities.getType()), DimensionHandlerFactory.class).buildHandler(dimensionName, capabilities)`

?